frontend/apps/reader/modules/readerwikipedia.lua

```diff
@@ -175,109 +174,21 @@ function ReaderWikipedia:addToMainMenu(menu_items)
             { -- setting used by dictquicklookup
                 text = _("Set Wikipedia 'Save as EPUB' folder"),
                 keep_menu_open = true,
-                callback = function()
-                    local choose_directory = function()
-                        -- Default directory as chosen by DictQuickLookup
-                        local default_dir = G_reader_settings:readSetting("wikipedia_save_dir")
-                                         or G_reader_settings:readSetting("home_dir")
-                                         or require("apps/filemanager/filemanagerutil").getDefaultDir()
-                        local dialog
-                        dialog = ButtonDialogTitle:new{
-                            title = T(_("Current Wikipedia 'Save as EPUB' folder:\n\n%1\n"), BD.dirpath(default_dir)),
-                            buttons = {
-                                {
-                                    {
-                                        text = _("Keep this folder"),
-                                        callback = function()
-                                            UIManager:close(dialog)
-                                        end,
-                                    },
-                                },
-                                {
-                                    {
-                                    text = _("Choose other folder"),
-                                    callback = function()
-                                        UIManager:close(dialog)
-                                        -- Use currently read book's directory as starting point,
-                                        -- so a user reading a wikipedia article can quickly select
-                                        -- it to save related new articles in the same directory
-                                        local dir = G_reader_settings:readSetting("wikipedia_save_dir")
-                                                 or G_reader_settings:readSetting("home_dir")
-                                                 or require("apps/filemanager/filemanagerutil").getDefaultDir()
-                                                 or "/"
```

Comment:
(Did I really not implement what the comment says ? `Use currently read book's directory as starting point` ?!
It's there as is since b46628c08cd7. Looks like I misplaced that comment, or I don't get any longer why I put it here...)

I admit all this here feels a bit twisted and ideas (and callbacks :) stacked as they were added :)
No feature lost or behaviour change (except for the wording and the final InfoMessage removed) with your simplifications? No ambiguity with the simplified UI?

Reply:
Maybe because we have a checkbox "Save Wikipedia EPUB in current book folder"?

Reply:
EPUB is saved to (in order of priority):
-book folder if "wikipedia_save_in_book_dir" is set
-"wikipedia_save_dir" if it is set
-default folder (Home/Wikipedia)
Absent folder is created.